packages/lib/src/interactions/utils.ts

```diff
@@ -31,17 +32,26 @@ export function getRatioEndPoint(
   const originalRatio =
     Math.abs(endPoint.x - startPoint.x) / Math.abs(endPoint.y - startPoint.y);
 
-  if (originalRatio < ratio) {
-    return new Vector2(
-      startPoint.x + widthSign * height * ratio,
-      startPoint.y + heightSign * height
-    );
+  const shiftX = widthSign * (originalRatio < ratio ? height * ratio : width);
+  const shiftY = heightSign * (originalRatio < ratio ? height : width / ratio);
+
+  if (!clampCenter) {
+    return [
+      startPoint,
+      new Vector2(startPoint.x + shiftX, startPoint.y + shiftY),
+    ];
```

Comment:
This is equivalent to the previous algorithm: the shift is applied fully to the `endPoint` while the `startPoint` stays at the same position ...